### PR TITLE
Moves entity save to before review call

### LINKED ISSUE #4879: Updating flagged RBItem status does not update the flagged state
**Context:** viewing a Reportback with a series of reviewed items in the CMS and one of them has a status of _flagged_, all of the items including the flagged one will have a `1` in the `flagged` field.

If you update the flagged item and change the status to promoted or approved, the `status` field updates properly, but the `flagged` field remains as `1` for all the items in the RB.

@mikefantini @namimody @angaither

### PR DESCRIPTION
**Pull request by itsjoekent**
The reason the bug was occuring and why this change fixes it is outlined in detail here: https://github.com/DoSomething/phoenix/issues/4879#issuecomment-134745880

Fixes #4879 

@angaither

### CODE REVIEW

File: lib/modules/dosomething/dosomething_reportback/includes/ReportbackItem.php [hunk 1]
```diff
@@ -182,6 +182,10 @@ public function review($values) {
     if (isset($values['promoted_reason'])) {
       $promoted_reason = $values['promoted_reason'];
     }
+
+    // Must be saved before calling setFlaggedPromoted
+    $updatedReportbackItem = entity_save('reportback_item', $this);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_reportback/includes/ReportbackItem.php]: javascript is invading your life! `snake_case`

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_reportback/includes/ReportbackItem.php]: oh my god it really is, i think i need to see a doctor

**Comment by angaither** [comment]
@deadlybutter I'm curious _why_ calling save on the item after the `setFlaggedPromoted` function doesn't allow this to work...

**Comment by itsjoekent** [comment]
sure! the problem is we're calling setFlaggedPromoted, which looks at all of the reportback items to determine if it should be flagged, promoted, etc. Because this item wasn't saved yet, that function still thinks it's flagged. Because this also happens to be the reportback item with the reason, the logic ignores it thinking we're supposed to flag.

Does that make sense...?

**Comment by angaither** [comment]
ahhh, got it thanks @deadlybutter

**Comment by itsjoekent** [comment]
@angaither good to go?

**Comment by angaither** [comment]
:+1:

**Pull request merged**